index.js

```diff
@@ -174,16 +177,43 @@ Object.defineProperty(File.prototype, 'contents', {
   },
 });
 
+Object.defineProperty(File.prototype, 'cwd', {
+  get: function() {
+    return this._cwd;
+  },
+  set: function(cwd) {
+    if (typeof cwd !== 'string') {
+      throw new Error('cwd should be a string.');
+    }
+    this._cwd = endWithSep(normalize(cwd));
+  },
+});
+
+Object.defineProperty(File.prototype, 'base', {
+  get: function() {
+    return this._base || this._cwd;
+  },
+  set: function(base) {
+    if (base && typeof base !== 'string') {
+      throw new Error('base should be a string.');
```

Comment:
Base actually can be a falsy value, which is how the setter switches to `null`, i.e. "default to `file.cwd`" mode. This means that the setter throws only when any truthy value except a string is passed, in which case the **base should be a string** is fitting. It can maybe changed to `base must be a string or falsy.`, but `non-empty string.` wouldn't be accurate.

But I added this message to `cwd` where it makes sense.